src/traces/scatterternary/plot.js

```diff
@@ -13,11 +13,16 @@ var scatterPlot = require('../scatter/plot');
 
 
 module.exports = function plot(ternary, data) {
+    var plotContainer = ternary.plotContainer;
+
+    // remove all nodes inside the scatter layer
+    plotContainer.select('.scatterlayer').selectAll('*').remove();
```

Comment:
This new line may look bad, but it's way better than previously.

Commit 39b8acf54814c6cb4b33ac797a5027637edf7e07 had a nasty side effect that took me quite some time to debug.

Previously, the ternary code stored a ref to a mock cartesian subplot object in `fullLayout._plots` which led to full framework redraws on every `restyle`. This was caused by this [block](https://github.com/plotly/plotly.js/blob/b5c98271d32647364cf46b338003a077ad781292/src/plot_api/plot_api.js#L-132L134) where `oldSubplot` was equal to `ternary` and the new `subplot` was equal to `''` as `Axes.getSubplots` only catches cartesian subplots. So **every** restyle call on ternary plots went through the slow [`makePlotFramework`](https://github.com/plotly/plotly.js/blob/b5c98271d32647364cf46b338003a077ad781292/src/plot_api/plot_api.js#L2587) routine.

We could definitely do better than removing all the inner nodes, but keeping the ordering of `lines`, `markers` and `text` nodes is tricky. At least now, `scatterternary` traces are on par with [`scattergeo`](https://github.com/plotly/plotly.js/blob/b5c98271d32647364cf46b338003a077ad781292/src/traces/scattergeo/plot.js#L129).